sway/tree/workspace.c

```diff
@@ -121,17 +127,8 @@ void next_name_map(struct sway_container *ws, void *data) {
 
 static bool workspace_valid_on_output(const char *output_name,
 		const char *ws_name) {
-	int i;
-	for (i = 0; i < config->workspace_outputs->length; ++i) {
-		struct workspace_output *wso = config->workspace_outputs->items[i];
-		if (strcasecmp(wso->workspace, ws_name) == 0) {
-			if (strcasecmp(wso->output, output_name) != 0) {
-				return false;
-			}
-		}
-	}
-
-	return true;
+	struct workspace_config *wsc = workspace_find_config(ws_name);
+	return !wsc || strcmp(wsc->output, output_name) == 0;
```

Comment:
It seems that in the future, `wsc->output` may be NULL. We probably need to check this here.